libbpf-tools/offcputime.h

```diff
@@ -0,0 +1,15 @@
+/* SPDX-License-Identifier: (LGPL-2.1 OR BSD-2-Clause) */
+#ifndef __OFFCPUTIME_H
+#define __OFFCPUTIME_H
+
+#define TASK_COMM_LEN		16
+
+struct key_t {
+	__u32 pid;
+	__u32 tgid;
+	int user_stack_id;
+	int kern_stack_id;
+	char comm[TASK_COMM_LEN]; /* in order to distinguish idle threads */
```

Comment:
The reason why it has not been moved to val is to distinguish between different core idle threads. I remember you mentioned that comparing char [] in key is slower. 

Since the id of idle is 0, I think we can set the pid field to the value of `bpf_get_smp_processor_id()`  instead for better performance?